server-src/Main.hs

```diff
@@ -88,45 +88,44 @@ app cache request respond = do
     (requestMethod request <> " " <> rawPathInfo request)
 
   case request of
-    GET [] -> do
-      indexHtml <- getDataFileName "index.html"
-      respond
-        (responseFile
-          status200
-          [(hContentType, "text/html")]
-          indexHtml
-          Nothing)
 
-    GET ["branch", branch] ->
+    GET ["api", "branch", branch] ->
       handleGetBranch cache branch >>= respond
 
-    GET ["declaration", declaration] ->
+
+    GET ["api", "declaration", declaration] ->
       handleGetDeclaration cache declaration >>= respond
 
-    GET ["head"] -> do
+    GET ["api", "head"] -> do
       [head] <- listDirectory ".unison/v1/paths/_head"
       respond
         (responseLBS
           status200
           [(hContentType, "application/json")]
           (LazyByteString.Char8.pack (show head)))
 
-    GET ["patch", patch] ->
+    GET ["api", "patch", patch] ->
       respond
         (responseFile
           status200
           [(hContentType, "application/octet-stream")]
           (".unison/v1/patches/" <> Text.unpack patch <> ".up")
           Nothing)
 
-    GET ["term", term, "term"] ->
+    GET ["api", "term", term, "term"] ->
       handleGetTerm cache term >>= respond
 
-    GET ["term", term, "type"] ->
-      handleGetTermType cache term >>= respond
+    GET ["api", "term", termType, "type"] ->
+      handleGetTermType cache termType >>= respond
 
-    _ ->
-      respond (responseLBS status404 [] "")
+    _ -> do
```

Comment:
What was the logic behind moving serving `index.html` in the fallthrough case instead of 404ing?

Reply:
The logic was that we would implement the 404 handler using the elm SPA routes instead of having to duplicate writing the routes again for the haskell local server. However right now we have 404s redirecting to the head hash; that's a mistake and I can set that up to 404 instead.

Reply:
hey elliot, need any help doing this? we can always merge and you can add this change later.